src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,153 @@
+  }
+
+  /** Judge current pixel's RGBA against original pixel's RGBA to see if it
+   *  can still be part of the mask (using tolerance criteria)
+   */
+  getIsStillMask(originalPixel: Array<number>, imgData: ImageData,
+                         pixelCoord: Array<number>, tolerance: number,
+                         visited: Set<number>): boolean {
+    let isStillMask: boolean = true;
+
+    // Preface; check if pixel is valid (indexing errs and repeat values)
+    let isValid: boolean = this.getIsValid(imgData.width, imgData.height, pixelCoord, visited);
+    if (!isValid) {
+      // Automatically not in mask b/c failed vailidity test
+      return !isStillMask;
+    }
+
+    // Get array of attributes of current pixel
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+    let curPixel: Array<number> = this.dataArrayToRGBA(imgData, curX, curY);
+
+    // All attributes of the pixel (R,G,B, and A) must be within tolerance level
+    for (let i = 0; i < 4; i++) {
+      let upperTolerance: boolean = curPixel[i] > originalPixel[i] + tolerance;
+      let lowerTolerance: boolean = curPixel[i] < originalPixel[i] - tolerance;
+      let failsTolerance: boolean = upperTolerance || lowerTolerance;
+      if (failsTolerance) {
+        return !isStillMask;
+      }
+    }
+
+    return isStillMask;
+  }
+
+  // Check if @pixelCoord is in bounds and makes sure it's not a repeat coord
+  getIsValid(imgWidth: number, imgHeight: number, pixelCoord: Array<number>, visited: Set<number>): boolean {
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+
+    let isValid: boolean = true;
+    // Check bounds of indexing
+    let yOutOfBounds: boolean = curY < 0 || curY > imgHeight - 1;
+    let xOutOfBounds: boolean = curX < 0 || curX > imgWidth - 1;
+    if (yOutOfBounds || xOutOfBounds) {
+      return !isValid;
+    }
+    // Do not push repeat coords to heap
+    let indexAsDataArray: number = this.coordToDataArrayIndices(curX, curY, imgWidth)[0];
+    if (visited.has(indexAsDataArray)) {
+      return !isValid;
+    }
+    visited.add(indexAsDataArray);
+
+    return isValid;
+  }
+
+  // Return pixel attributes of @imgData at [@xCoord, @yCoord] as Array<number>
+  dataArrayToRGBA(imgData: ImageData, xCoord: number, yCoord: number): Array<number> {
+    // Unpack imgData for readability
+    let data: Uint8ClampedArray = imgData.data;
+    let imgWidth: number = imgData.width;
+
+    // Pixel attributes in imgData are organized adjacently in a 1-D array
+    let pixelIndex: number = (xCoord + (imgWidth * yCoord)) * 4;
+    let red: number = data[pixelIndex];
+    let green: number = data[pixelIndex + 1];
+    let blue: number = data[pixelIndex + 2];
+    let alpha: number = data[pixelIndex + 3];
+    // Store original pixel's attributes
+    let pixel: Array<number> = [red, green, blue, alpha];
+    return pixel;
+  }
+
+  // Convert coord [@x, @y] (2-D) to indexing style of DataArray (1-D)
```

Comment:
From the comment I didn't realize you were returning the RGBA indices.  I thought it as just the start of the pixel so it would be good to clarify that.